qiita_ware/ebi.py

```diff
@@ -362,8 +372,10 @@ def generate_sample_xml(self):
             'xmlns:xsi': self.xmlns_xsi,
             "xsi:noNamespaceSchemaLocation": self.xsi_noNSL % "sample"})
 
-        for sample_name, sample_info in sorted(viewitems(self.samples)):
-            sample_info = dict(sample_info)
+        samples = samples if samples is not None else viewkeys(self.samples)
```

Comment:
This is a rather subjective comment, but this line doesn't need to be a one-liner, it should jus be an if statement:

```python
if samples is not None:
    samples = viewkeys(self.samples)
```